cypress/e2e/multiplot.cy.ts

```diff
@@ -0,0 +1,76 @@
+import { parseHTML } from '../support/parseHTML';
+import multiplotData from '../data_src/multiplot.data.json';
+import primitiveGroupContainerData from '../data_src/primitivegroupcontainer.data.json';
+
+const featureName = "multiplot"
+
+before(() => {
+  parseHTML(featureName, multiplotData)
+})
+
+describe('MULTIPLOT CANVAS', function () {
+  const describeTitle = this.title + ' -- '
+  beforeEach(() => {
+    cy.visit("cypress/html_files/" + featureName + ".html");
+    cy.wait(500)
+  })
+
+  it("should draw canvas", function () {
+    cy.compareSnapshot(describeTitle + this.test.title, 0.05);
+  })
+
+  it("should remove primitive group from container in multiplot", function () {
+    cy.window().then((win) => {
+      win.eval('multiplot').remove_all_primitive_groups_from_container(6);
+      cy.compareSnapshot(describeTitle + this.test.title, 0.05);
+    })
+  })
+
+  it("should add primitive group container in multiplot", function () {
+    cy.window().its('multiplot').then((multiplot) => {
+      multiplot.add_primitive_group_container(primitiveGroupContainerData, [], null)
+      cy.wrap('multiplot').as('multiplot')
+      cy.compareSnapshot(describeTitle + this.test.title, 0.05);
+    })
+  })
+
+  it("should reorder all plots in canvas", function () {
+    cy.window().then((win) => {
+      let multiplot = win.eval('multiplot')
+      multiplot.add_primitive_group_container(primitiveGroupContainerData, [], null)
+      cy.wait(500)
```

Comment:
As noted in other comments, these wait could be really costly. For example if there are 10 tests like this and we tests several browser, this test only would last 30s by browser

Reply:
Removed, it was here to comfort with some results

Reply:
If this is for dev purpose, there probably is a way to perform only certain action while on dev configuration vs being on prod

Reply:
It was more to be sure that cypress was running the normal way.  Now that Cypress is configured, I guess it is useless unless it is required for the test to be truthy (in that case, the wait would be the normal way to write the test because of an asynchronous behavior=)

Reply:
I think this would be a dangerous way to do it. Cypress should normally have features that enable asynchronous mocking

Reply:
I know that sometimes, the browser can be long to compute results (for example, if you compute something in 10 seconds) and Cypress does not wait for it to be completed before 'snapshooting' so the test is always False unless you set a cy.wait after the command line

Reply:
Yeah, that is why the wait is an unsafe way to handle ansychronous stuff, because it could be 10s, 20s, 2 minutes , a day or 3 years , so waiting for a result is considered unsafe. Cypress should have features to wait for an operation to finish because that is how HTTP protocol works